### PR TITLE
Add host metrics.

### PR DESCRIPTION
**Pull request by CyrusRoshan**
Not e2e tested yet.

### CODE REVIEW

File: pkg/agent/service/client/client.go [hunk 1]
```diff
@@ -7,10 +7,24 @@ import (
 	"net/http"
 )
 
-func GetDeviceMetrics(deviceConn net.Conn) (*http.Response, error) {
+func GetDeviceAgentMetrics(deviceConn net.Conn) (*http.Response, error) {
```
Comment (joshwget) [inline pkg/agent/service/client/client.go]: Can we drop "Device" from this name? I don't know why I added it in the first place. It's pretty redundant.

File: pkg/agent/service/client/client.go [hunk 2]
```diff
@@ -7,10 +7,24 @@ import (
...
+	)
+
+	if err := req.Write(deviceConn); err != nil {
+		return nil, err
+	}
+
+	return http.ReadResponse(bufio.NewReader(deviceConn), req)
+}
+
+func GetDeviceHostMetrics(deviceConn net.Conn) (*http.Response, error) {
```
Comment (joshwget) [inline pkg/agent/service/client/client.go]: Same here

File: pkg/agent/metrics/metrics.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/node_exporter/collector"
+)
+
+func HostMetricsHandler() (*http.Handler, error) {
+	maxRequests := 3
```
Comment (joshwget) [inline pkg/agent/metrics/metrics.go]: Can we rename to "maxRequestsInFlight" and make it a constant at the top? A comment would be nice explaining where the 3 came from.

**Comment by joshwget** [comment]
I think we need to run `go mod vendor`

File: pkg/agent/service/client/client.go (hunk 1 above)
Comment (CyrusRoshan) [reply pkg/agent/service/client/client.go]: I'll consider it, after I change it to the much-more-readable `GetDeviceAgentMetricsInOpenMetricsFormat`.

**Pull request merged**